### PR TITLE
Add apt-get update and upgrade to update_documentation action

### PR DESCRIPTION
**Pull request by holke**
**_Describe your changes here:_**

Add apt-get update and upgrade to update_documentation action.

The action was failing due to the doxygen package not being installed properly.
This may fix it.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

**Review by lukasdreyer — APPROVED** [review]

**Review by sandro-elsweijer — APPROVED** [review]

File: .github/workflows/update_documentation.yml [hunk 1]
```diff
@@ -45,6 +45,8 @@ jobs:
 # Setup and bootstrap
 #
     - uses: actions/checkout@v4
```
Comment (holke) [inline .github/workflows/update_documentation.yml:47]:
```suggestion
    - uses: actions/checkout@v4
       with: 
        fetch-depth: 0
```

File: .github/workflows/update_documentation.yml [hunk 2]
```diff
@@ -45,6 +45,10 @@ jobs:
...
+       with:
```
Comment (holke) [inline .github/workflows/update_documentation.yml:48]:
```suggestion
      with: 
```

File: .github/workflows/tests_t8code_serial.yml [hunk 3]
```diff
@@ -63,7 +63,7 @@ jobs:
       with: 
         fetch-depth: 0
     - name: Update packages
-      run: apt-get update && apt-get upgrade -y
+      run: sudo apt-get update && sudo apt-get upgrade -y
```
Comment (holke) [inline .github/workflows/tests_t8code_serial.yml:66]:
```suggestion
      run: apt-get update && apt-get upgrade -y
```

File: .github/workflows/update_documentation.yml [hunk 4]
```diff
@@ -45,6 +45,10 @@ jobs:
 # Setup and bootstrap
 #
     - uses: actions/checkout@v4
+      with: 
+        fetch-depth: 0
+    - name: Update packages
+      run: apt-get update && apt-get upgrade -y
```
Comment (holke) [inline .github/workflows/update_documentation.yml:51]:
```suggestion
      run: sudo apt-get update && sudo apt-get upgrade -y
```

File: .github/workflows/update_documentation.yml [hunk 5]
```diff
@@ -45,6 +45,10 @@ jobs:
 # Setup and bootstrap
 #
     - uses: actions/checkout@v4
+      with: 
+        fetch-depth: 0
```
Comment (lukasdreyer) [inline .github/workflows/update_documentation.yml:49]: What does this do? Was it actually necessary for the fix?

**Review by lukasdreyer — COMMENTED** [review]
Thanks for the quick fix, only a small question

File: .github/workflows/update_documentation.yml (hunk 5 above)
Comment (lukasdreyer) [reply .github/workflows/update_documentation.yml:49]: Set fetch-depth: 0 to fetch all history for all branches and tags

I would think that we only need the history for the current branch. Actually I am not sure if we need them in any of our workflows

**Pull request merged**